sys/shell/cmds/vfs.c

```diff
@@ -720,6 +720,91 @@ static int _vfs_handler(int argc, char **argv)
 
 SHELL_COMMAND(vfs, "virtual file system operations", _vfs_handler);
 
+#if MODULE_SHELL_CMD_GENFILE
+static char _get_char(unsigned i)
+{
+    i %= 2 * (1 + 'z' - 'a') + 10;
+
+    if (i < 10) {
+        return '0' + i;
+    }
+    i -= 10;
+
+    if (i <= 'z' - 'a') {
+        return 'a' + i;
+    }
+    i -= 1 + 'z' - 'a';
+
+    return 'A' + i;
+}
+
+static void _write_block(int fd, unsigned bs, unsigned i)
+{
+    char block[bs];
+    char *buf = block;
+
+    buf += snprintf(buf, bs, "|%03u|", i);
+
+    memset(buf, _get_char(i), &block[bs] - buf);
+    block[bs - 1] = '\n';
+
+    vfs_write(fd, block, bs);
+}
+
+static int _vfs_genfile_cmd(int argc, char **argv)
+{
+    unsigned blocksize = 64;
+    unsigned blocks = 32;
+    int fd = STDOUT_FILENO;
+
+    const char *cmdname = argv[0];
+    while (argc > 1 && argv[1][0] == '-') {
+        char *optarg = argc > 2 ? argv[2] : NULL;
+        char opt = argv[1][1];
+
+        if (optarg == NULL) {
+            printf("missing argument\n");
+            opt = '?';
+        }
+
+        switch (opt) {
+        case '?':
+            printf("usage: %s [-o <file>] [-b <block size>] [-n num blocks]\n",
+                   cmdname);
+            return 0;
+        case 'o':
+            fd = vfs_open(optarg, O_CREAT | O_TRUNC | O_WRONLY, 0644);
+            break;
+        case 'b':
+            blocksize = atoi(optarg);
+            break;
+        case 'n':
+            blocks = atoi(optarg);
+            break;
+        default:
+            printf("unknown option '%s'\n", argv[1]);
+            return 1;
+        }
+        argc -= 2;
+        argv += 2;
+    }
+
+    if (fd < 0) {
+        return fd;
+    }
+
+    for (unsigned i = 0; i < blocks; ++i) {
+        _write_block(fd, blocksize, i);
+    }
+
+    if (fd != STDOUT_FILENO) {
+        vfs_close(fd);
+    }
+    return 0;
```

Comment:
Maybe print how many bytes got written? Eg, `genfile -o out.bin -b 32 -n 32` -> `1024 byte(s) written.`. In my opinion a nice UX.